spec/helpers.lua

```diff
@@ -1585,12 +1586,27 @@ end
 -- 
 -- NOTE: this function is not available for DBless-mode
 -- @function wait_for_all_config_update
--- @tparam[opt=30] number timeout maximum time to wait
--- @tparam[opt] number admin_client_timeout, to override the default timeout setting
--- @tparam[opt] number forced_admin_port to override the default port of admin API
+-- @tparam[opt=30] table opt a table contains params
+-- 
+-- timeout: maximum seconds to wait, defatuls is 30
+-- 
+-- admin_client_timeout: to override the default timeout setting
+-- 
+-- forced_admin_port: to override the default Admin API port
+-- 
+-- proxy_client_timeout: to override the default timeout setting
+-- 
+-- forced_proxy_port: to override the default proxy port
+-- 
 -- @usage helpers.wait_for_all_config_update()
-local function wait_for_all_config_update(timeout, admin_client_timeout, forced_admin_port)
-  timeout = timeout or 30
+local function wait_for_all_config_update(opt)
+  opt = opt or {}
+
+  local timeout = opt.timeout or 30
+  local admin_client_timeout = opt.admin_client_timeout
+  local forced_admin_port = opt.forced_admin_port
+  local proxy_client_timeout = opt.proxy_client_timeout
+  local forced_proxy_port = opt.forced_proxy_port
```

Comment:
Use proper ldoc comments and use `opts` instead of `opt`.
```suggestion
-- @tparam[opt] table opts a table contains params
-- @tparam[opt=30] number timeout maximum seconds to wait, defatuls is 30
-- @tparam[opt=30] number admin_client_timeout to override the default timeout setting
-- @tparam[opt] number forced_admin_port to override the default Admin API port
-- @tparam[opt=30] number proxy_client_timeout to override the default timeout setting
-- @tparam[opt] number forced_proxy_port to override the default proxy port
-- @usage helpers.wait_for_all_config_update()
local function wait_for_all_config_update(opts)
  opts = opts or {}

  local timeout = opts.timeout or 30
  local admin_client_timeout = opts.admin_client_timeout
  local forced_admin_port = opts.forced_admin_port
  local proxy_client_timeout = opts.proxy_client_timeout
  local forced_proxy_port = opts.forced_proxy_port
```

Reply:
done